src/flowsynth.py

```diff
@@ -466,25 +465,56 @@ def _valid_mac(self, mac):
     #This function expects all inputs to be enclosed within double quotes
     def parse_content(self, content):
         """ parse and render a content keyword """
+
+        # this regex is somewhat gnarly but leaving for now ... (why not
+        # just strip off double quotes on ends)?
         pcre_text = r'"([^\\"]*(?:\\.[^\\"]*)*)"'
 
+        result = bytearray()
 
         #first, check for text
         mo_text = re.match(pcre_text, content)
         if (mo_text != None):
-            logging.debug("Content: %s", mo_text.group(1))
-
             content_text = mo_text.group(1)
-            replacements = re.findall(r"\\x[a-fA-F0-9]{2}", content_text)
-            for replacement in replacements:
-                content_text = content_text.replace(replacement, chr(int(replacement[2:], 16)))
+            logging.debug("Content: %s (length %d)" % (content_text, len(content_text)))
+            start = 0
+            previous_end = 0
+            # Flowsynth supports encoding arbitrary bytes with the "\xHH" notation where "HH" is
+            # the hexidecimal representation of the byte. That is what is handled here, while
+            # maintaining the rest of the content data as UTF-8.
```

Comment:
If you're OK with all of Python's string escaping (not just hex escapes), I think you can replace this entire block with

```python
result.extend(context_text.decode('unicode_escape'))
```

That will work in both Python 2:

```python
>>> r'123\x34'
'123\\x34'
>>> r'123\x34'.decode('unicode_escape')
u'1234'
```

and 3:
```python
>>> rb'123\x34'
b'123\\x34'
>>> rb'123\x34'.decode('unicode_escape')
'1234'
```

As you can see, it works with byte strings.  I can't tell from reading the code if `content_text` is unicode.  If so, you'll want to decode it into backslash-escaped ASCII first:

```python
>>> u'123\\x34\0'.encode('ascii', 'backslashreplace').decode('unicode_escape')
u'1234\x00'
```